tests/test_simpleButler.py

```diff
@@ -87,105 +93,14 @@ def makeRegistry(self) -> Registry:
-        datastore: `Datastore`
-            The `Datastore` instance or mock that datasets were loaded into.
-        """
-        if registry is None:
-            registry = self.makeRegistry()
-        if datastore is None:
-            datastore = unittest.mock.Mock(spec=Datastore)
-        if isinstance(file, str):
-            with open(file, 'r') as stream:
-                backend = YamlRepoImportBackend(stream, registry)
-                backend.register()
-                backend.load(datastore, directory=directory, transfer=transfer,
-                             skip_dimensions=skip_dimensions)
-        else:
-            backend = YamlRepoImportBackend(file, registry)
-            backend.register()
-            backend.load(datastore, directory=directory, transfer=transfer, skip_dimensions=skip_dimensions)
-        return (registry, datastore)
-
-    @contextlib.contextmanager
-    def runExport(self, *,
-                  registry: Optional[Registry] = None,
-                  datastore: Optional[Datastore] = None,
-                  stream: Optional[TextIO] = None,
-                  directory: Optional[str] = None,
-                  transfer: Optional[str] = None) -> Generator[RepoExportContext, None, None]:
-        """Export repository data to an export file.
-
-        Parameters
-        ----------
-        registry: `Registry`, optional
-            Registry instance to load datasets from.  If not provided,
-            `makeRegistry` is called.
-        datastore: `Datastore`, optional
-            Datastore instance to load datasets from (may be a mock).  If not
-            provided, a mock is created.
-        directory: `str`, optional
-            Directory to contain exported file.  Ignored if ``datastore`` is
-            a mock.
-        stream : `TextIO`, optional
-            Writeable file-like object pointing at the export file to write.
-        transfer: `str`, optional
-            Transfer mode.  See `Datastore.ingest`.
-
-        Yields
-        ------
-        context : `RepoExportContext`
-            A helper object that can be used to export repo data.  This is
-            wrapped in a context manager via the `contextlib.contextmanager`
-            decorator.
-        """
-        if registry is None:
-            registry = self.makeRegistry()
-        if datastore is None:
-            datastore = unittest.mock.Mock(spec=Datastore)
-            datastore.export = _mock_export
-        backend = YamlRepoExportBackend(stream)
-        try:
-            helper = RepoExportContext(registry, datastore, backend=backend,
-                                       directory=directory, transfer=transfer)
-            yield helper
-        except BaseException:
-            raise
-        else:
-            helper._finish()
+    def makeButler(self, **kwargs: Any) -> Butler:
+        config = ButlerConfig()
+        config["registry", "db"] = "sqlite://"
```

Comment:
It might be clearer here if you use `:memory:` explicitly here like is done in https://github.com/lsst/daf_butler/blob/master/python/lsst/daf/butler/tests/_testRepo.py#L76